src/libraries/System.Private.CoreLib/src/System/Resources/ResourceSet.cs

```diff
@@ -134,59 +124,46 @@ IEnumerator IEnumerable.GetEnumerator()
 
         private IDictionaryEnumerator GetEnumeratorHelper()
         {
-            Hashtable? copyOfTable = Table;  // Avoid a race with Dispose
-            if (copyOfTable == null)
-                throw new ObjectDisposedException(null, SR.ObjectDisposed_ResourceSet);
-            return copyOfTable.GetEnumerator();
+            // Avoid a race with Dispose
+            return (_table as IDictionary)?.GetEnumerator() ?? throw new ObjectDisposedException(null, SR.ObjectDisposed_ResourceSet);
```

Comment:
I do not think that the cast to `IDictionary` here is necessary. `Dictionary.GetEnumerator()` and `IDictionary.GetEnumerator()` on `Dictionary` are returning the exact same enumerator.

Reply:
Ah, ok. I see how this works. The enumerator is configured differently for IDictionary.

Reply:
Yep